src/storage/txn/actions/shared.rs

```diff
@@ -0,0 +1,65 @@
+use crate::storage::mvcc::metrics::CONCURRENCY_MANAGER_LOCK_DURATION_HISTOGRAM;
+use crate::storage::mvcc::{Lock, LockType, MvccTxn, Result as MvccResult, TimeStamp};
+use crate::storage::Snapshot;
+use std::cmp;
+use txn_types::{is_short_value, Key, Value};
+
+pub fn prewrite_key_value<S: Snapshot>(
```

Comment:
Why is it put in `shared`? What is a "shared" action?

Reply:
> Why is it put in `shared`? What is a "shared" action?

This is shared by both `prewrite` and `pessimistic_prewrite`.

I plan to put things shared by different actions here.

Or do you think `prewrite_key_value` should be an action on it own?

Reply:
OK, I see. I thought `prewrite` and `pessmistic_prewrite` were all in the same action.

Reply:
I don't know what the best solution is, but I'm a bit worried that "shared" will finally become a "hodgepodge".

Reply:
prewrite and pessimistic prewrite are the same request, but get converted into different commands. I think a better way to do it (long term, not here) is for there to be one command and we separate out the action code (thus keeping a 1:1 mapping from requests to commands).

I think somehow there is a missing abstraction here for some (or most or all) of the shared code, I think it is worth experimenting and seeing where it leads, because I don't think there have been any good proposals so far.